Client/Frontend/Home/FxHomeJumpBackInView.swift

```diff
@@ -0,0 +1,251 @@
+
+        NSLayoutConstraint.activate([
+            collectionView.topAnchor.constraint(equalTo: contentView.topAnchor, constant: 2),
+            collectionView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            collectionView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -2),
+            collectionView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor)
+        ])
+    }
+}
+
+extension FxHomeJumpBackInCollectionCell: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        viewModel.updateDataAnd(layoutVariables)
+        return viewModel.jumpableTabs.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: JumpBackInCell.cellIdentifier, for: indexPath) as! JumpBackInCell
+
+        if let item = viewModel.jumpableTabs[safe: indexPath.row] {
+            let itemURL = item.url?.absoluteString ?? ""
+            let site = Site(url: itemURL, title: item.displayTitle, bookmarked: true)
+
+            profile?.favicons.getFaviconImage(forSite: site).uponQueue(.main, block: { result in
+                guard let image = result.successValue else { return }
+                cell.heroImage.image = image
+                cell.setNeedsLayout()
+            })
+
+            cell.itemTitle.text = site.title
+            // TODO: Determine source string here, if from synced tabs. Simply update `cell.itemDetails.text`
+        }
+
+        return cell
+    }
+}
+
+extension FxHomeJumpBackInCollectionCell: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        if let tab = viewModel.jumpableTabs[safe: indexPath.row] {
+            viewModel.switchTo(tab)
+        }
+    }
+}
+
+extension FxHomeJumpBackInCollectionCell: UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+
+        var itemWidth: CGFloat
+        let totalHorizontalSpacing = collectionView.bounds.width
+        let columns = layoutVariables.columns
+        if columns == 2 {
+            itemWidth = (totalHorizontalSpacing - JumpBackInCollectionCellUX.iPadHorizontalSpacing) / columns
+        } else {
+            itemWidth = totalHorizontalSpacing / columns
+        }
+        let itemSize = CGSize(width: itemWidth, height: JumpBackInCollectionCellUX.cellHeight)
+
+        return itemSize
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        if UIDevice.current.userInterfaceIdiom == .phone {
```

Comment:
nit: Similar to `deviceIsiPad`, would it be okay to create something similar for iPhone for consistency? And maybe we can implement those as extensions on UIDevice?

Reply:
Or, since most of the codebase uses `UIDevice.current.userInterfaceIdiom == .something` - it's up to you to continue that or shorten it.